Backend/database/repositories/filesystem.go

```diff
@@ -3,14 +3,87 @@ package repositories
 
 import (
 	"errors"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
 	"strings"
+
+	"github.com/docker/docker/client"
 )
 
+const volume_path = "/var/lib/documents/data"
+
 // Implements IRepositoryInterface
 type FilesystemRepository struct {
 	embeddedContext
 }
 
+type DockerFileystemRepository struct {
+	cli *client.Client
+}
+
+// Create instance of DockerFileSystemRepository struct
+func NewDockerFilesystemRespository() (c *DockerFileystemRepository, err error) {
+	c = new(DockerFileystemRepository)
```

Comment:
nitpick: DockerFilesystemRepository is a struct type so it doesn't have to be initialized with ```new()```, instead you can just allocate it locally and return a pointer to it, during the escape analysis phase of compilation the compiler will automatically mark the variable as allocated on the heap for you:

```go
func myFunc() *DockerFilesystemRepository {
    myStruct := DockerFilesystemRepository {}
    return &myStruct
}

```